bgpd/bgp_route.c

```diff
@@ -11351,8 +11353,18 @@ static int bgp_table_stats_walker(struct thread *t)
 	return 0;
 }
 
-static int bgp_table_stats(struct vty *vty, struct bgp *bgp, afi_t afi,
-			   safi_t safi, struct json_object *json_array)
+static void bgp_table_stats_all(struct vty *vty, afi_t afi, safi_t safi,
```

Comment:
Function below return function is not being checked in all the caller function for this. It can happen that someone is trying to issue command for VRF which is not present and that can lead to BGP being null. In that case we end up printing all?

	bgp_vty_find_and_parse_afi_safi_bgp(vty, argv, argc, &idx, &afi, &safi,
					    &bgp, false);

Reply:
I think function which are calling function bgp_vty_find_and_parse_afi_safi_bgp are looking for idx and if it is zero then do a early return as below. I guess same should be followed for "DEFUN(show_ip_bgp_statistics_all, show_ip_bgp_statistics_all_cmd" such that it handles vrf "all".

	bgp_vty_find_and_parse_afi_safi_bgp(vty, argv, argc, &idx, &afi, &safi,
					    &bgp, false);
	if (!idx)
		return CMD_WARNING;

Reply:
>Function below return value is not being checked in all the caller functions. It can so happen that someone is trying to issue command for VRF which is not present and that can lead to BGP pointer being null. In that case we end up printing all which is not intended right?

Well, yes. Do you suggest checking for _all_ instead of bgp==0x0?

Reply:
Maybe we should return vrf name in `bgp_vty_find_and_parse_afi_safi_bgp` for further tests like this one.

Reply:
Yes. I think that is correct.

Reply:
Ok, I'll do the changes with vrf name.

Reply:
Actually, undefined vrf is already handled properly:

```
exit1-debian-9# sh ip bgp vrf belekas statistics
View/Vrf belekas is unknown
exit1-debian-9# 
```

Reply:
So, basically it's fine checking only bgp==0x0.

Reply:
Like we discussed offline "show ip bgp vrf all statistics-all " this should apperently display for all BGP session but it will do nothing.